GOVERNANCE.md

```diff
@@ -0,0 +1,477 @@
+as possible_ on the general development of the project.
+Releases should _never_ stop, prevent or even just slow down the normal
+development of the project.
+For example, halting development on the `main` branch and forbidding
+contributors and maintainers to merge changes to it even for as little as one
+hour just so that a new release can be published is not acceptable.
+Also should "patch" releases (sometimes called "bugfix releases") have no impact
+on the project pace or even concern developers - they should just happen in
+parallel without any influence on the normal development activities.
+
+With these requirements, we implement the following workflow.
+
+### Semver conformity and API coverage
+
+We adhere to [semver](https://semver.org).
+
+It has to be noted, though, that we also follow the
+[exception in semver compatibility as defined by cargo](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio),
+that specifically means:
+
+> While SemVer says there is no compatibility before 1.0.0, Cargo considers
+> 0.x.y to be compatible with 0.x.z, where y ≥ z and x > 0.
+
+Parts of the API that are covered by semver are explicitly marked. Our public
+API is extensively tested in integration tests. If changes to the public API of
+thin-egde.io are necessary, these changes are documented in the `CHANGELOG.md`
+file of the project, with reasoning why the change was necessary and how users
+can upgrade to the new interface(s).
+
+<!-- TODO: Expand on how exactly these changes are documented? -->
+
+We also define the following rules for our release process:
+
+* We release new features by increasing the "minor" version
+* When we release new patch releases, these releases never contain new features,
+  but only fixes to bugs.
+
+### Release cycle
+
+Minor releases are published roughly every TBD months.
+
+Until a minor release is published, the previous minor release _can_ receive
+patch releases. These patch releases _only_ contain bugfixes (accordingly to
+[semver](https://semver.org/)) and never contain new functionality.
+
+### Strategy
+
+We use the release-branch strategy, where development happens on the `main`
+branch and releases are done on dedicated `release-X.Y.Z` branches.
+
+Because we employ the "evergreen main branch" strategy and all of our merge
+commits on the `main` branch are guaranteed to successfully build and pass CI, a
+release can potentially be made starting from every commit on the `main` branch.
+
+For a description of the release workflow, read below.
+
+### Release maintainer
+
+For every release, one or more "release maintainers" are appointed. These
+maintainers have to be in the "release-maintainers" team.
+
+Multiple release maintainers should be prefered, to prevent release delay due to
+vacation, sickness or similar things.
+
+The responsibilities of the release maintainers are:
```

Comment:
Maybe we can boil this down to something like

```suggestion
For every release, one or more "release maintainers" are appointed by the core team.

The responsibilities of the release maintainers are:
```

That there should be multiple should be rather clear, right? And that they're appointed by the core team is also obvious, because they delegate who does what.

---

I'm not 100% sure on this suggestion here though and would welcome some ideas!

Reply:
I'm happy with this suggestion and applied it. Sometimes adding to much details make the reader miss the key points: here the need for release mainteners.